File: src/main/java/net/liyze/basin/core/scan/ListCommand.java
package net.liyze.basin.core.scan;

import com.itranswarp.summer.annotation.Component;
import net.liyze.basin.core.Basin;
import net.liyze.basin.core.Command;
import org.jetbrains.annotations.NotNull;

import java.util.List;
import java.util.Map;

import static net.liyze.basin.core.Basin.LOGGER;

/**
 * Print all command loaded.
 *
 * @author Liyze09
 */
@Component
public class ListCommand implements Command {
    @Override
    public void run(@NotNull List<String> args) {
        LOGGER.info("Commands");
        for (String i : Basin.commands.keySet()) {
            System.out.println(i);
        }
        LOGGER.info("Variables");
        for (Map.Entry<String, Object> i : Basin.envMap.entrySet()) {
            System.out.print(i.getKey() + " = ");
            System.out.println(i.getValue());
        }
    }

    @Override
    public @NotNull String Name() {
        return "list";
    }
}

File: src/main/kotlin/net/liyze/basin/core/scan/ListCommand.kt
/*
 * Copyright (c) 2023 Liyze09
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package net.liyze.basin.core.scan

import net.liyze.basin.Article
import net.liyze.basin.core.Command
import net.liyze.basin.core.commands
import net.liyze.basin.core.envMap
import org.slf4j.Logger
import java.io.PrintWriter
import java.io.StringWriter

/**
 * Print all command loaded.
 *
 * @author Liyze09
 */

class ListCommand : Command {
    override fun run(args: List<String?>, logger: Logger, context: Article) {
        val writer = StringWriter()
        val print = PrintWriter(writer)
        print.println()
        print.println("Commands")
        for (i in commands.keys) {
            print.print(" ")
            print.println(i)
        }
        print.println("Variables")
        for ((key, value) in envMap.entries) {
            print.print(" $key = ")
            print.println(value)
        }
        logger.info(writer.toString())
    }

    override fun getName(): String {
        return "list"
    }
}
